app/oracle-server/src/main/scala/org/bitcoins/oracle/server/OracleRoutes.scala

```
@@ -0,0 +1,116 @@
+
+                val (attestationJson, signatureJson) = event match {
+                  case completedEvent: CompletedEvent =>
+                    (Str(completedEvent.attestation.hex),
+                     Str(completedEvent.signature.hex))
+                  case _: PendingEvent =>
+                    (Str(""), Str(""))
+                }
+
+                val json = Obj(
+                  mutable.LinkedHashMap[String, Value](
+                    "nonce" -> Str(event.nonce.hex),
+                    "label" -> Str(event.label),
+                    "numOutcomes" -> Num(event.numOutcomes.toDouble),
+                    "signingVersion" -> Str(event.signingVersion.toString),
+                    "attestation" -> attestationJson,
+                    "signature" -> signatureJson,
+                    "outcomes" -> outcomesJson
+                  )
+                )
+                Server.httpSuccess(json.render(indent = 2))
+              case None =>
+                Server.httpSuccess("[]")
+            }
+          }
+      }
+
+    case ServerCommand("signevent", arr) =>
+      SignEvent.fromJsArr(arr) match {
+        case Failure(exception) =>
+          reject(ValidationRejection("failure", Some(exception)))
+        case Success(SignEvent(nonce, outcome)) =>
+          complete {
+            oracle.signEvent(nonce, outcome).map { sig =>
+              Server.httpSuccess(sig.hex)
+            }
+          }
+      }
+
+    case ServerCommand("getsignature", arr) =>
+      GetEvent.fromJsArr(arr) match {
+        case Failure(exception) =>
+          reject(ValidationRejection("failure", Some(exception)))
+        case Success(GetEvent(nonce)) =>
+          complete {
+            oracle.getEvent(nonce).map {
+              case Some(completed: CompletedEvent) =>
+                Server.httpSuccess(completed.signature.hex)
+              case None | Some(_: PendingEvent) =>
+                Server.httpSuccess("[]")
```

Probably should separate these cases return more useful messages in these cases.

---

Was trying to follow #1949 

Imo for apis like this we don't really want human readable returns

---

Maybe instead we should return an error